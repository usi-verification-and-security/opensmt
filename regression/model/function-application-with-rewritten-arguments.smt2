(set-logic QF_UF)
(declare-sort u 0)
(declare-fun f (u u) u)
(declare-fun b () Bool)
(declare-fun v6 () u)
(declare-fun v5 () u)
(declare-fun v4 () u)
(declare-fun v3 () u)
(declare-fun v1 () u)
(declare-fun v2 () u)

(assert (= v3 (f v5 v1)))
(assert (= v2 (f v3 v3)))
(assert (= v3 v6))
(assert (= b (and (= v5 v4) (= v5 v1))))
(assert b)

(check-sat)
(get-model)
