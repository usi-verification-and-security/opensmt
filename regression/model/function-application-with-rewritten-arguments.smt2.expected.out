sat
(
  (define-fun f ((x0 u) (x1 u)) u
    (ite (= (as @2 u) x0) (ite (= (as @2 u) x1) (as @3 u) (as @d2 u)) (ite (= (as @3 u) x0) (ite (= (as @3 u) x1) (as @4 u) (as @d2 u)) (as @d2 u))))
  (define-fun b () Bool
    true)
  (define-fun v6 () u
    (as @3 u))
  (define-fun v5 () u
    (as @2 u))
  (define-fun v4 () u
    (as @2 u))
  (define-fun v3 () u
    (as @3 u))
  (define-fun v1 () u
    (as @2 u))
  (define-fun v2 () u
    (as @4 u))
)
